### ISSUE TITLE
Network Manager dependency

### ISSUE THREAD

**Issue by alex-miro** [Azure/WALinuxAgent#864]
Starting at least with 2.2.14 wagon depends on NetworkManager. 
There are cases when NM interferes with the correct operations and is removed. 
What is the real reason for relying on NM ? In our case the system seems to function Ok without it (waagent version 2.2.15); just error messages in logs about missing NM. 
Request to make NM optional.

**Comment by brendandixon** [comment]
@alex-miro I have just verified in the code that only RedHat and similar images (e.g., CentOS) require Network Manager. And it has for quite some time now (the support was first added over two years ago). We did make a change about 7 months ago (see #573) altering how we set the `hostname`, but other than that, this code has changed.

It may be that you are comparing our latest branch (v.2.1+) with the older branch (v2.0.x). The move to v2.1+ rewrote significant portions of the agent, gaining many features and much improved distribution support.

I hope this helps.
